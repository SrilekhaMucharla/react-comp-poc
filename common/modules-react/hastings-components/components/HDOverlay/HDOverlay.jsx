import React from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes, css } from 'styled-components';

const fadeIn = (toOpacity) => keyframes`
    from {
        opacity: 0;
    }

    to {
        opacity: ${toOpacity};
    }
`;

const Overlay = styled.div`
    position: fixed;
    width: 100vw;
    height: 100vh;
    background: ${(props) => props.color};
    top: 0;
    left: 0;
    overflow: hidden;
    z-index: 1000004;
    opacity: ${(props) => props.opacity};
    animation: ${(props) => css`${props.animationDuration}s ${fadeIn(props.opacity)} ease-in`};
`;

const HDOverlay = (props) => {
    const { bgColor, opacity, animationDuration } = props;

    React.useEffect(() => {
        const previousStyle = { ...document.body.style };
        document.body.style.overflow = 'hidden';

        return () => {
            document.body.style = previousStyle;
        };
    }, []);

    return <Overlay color={bgColor} opacity={opacity} animationDuration={animationDuration} />;
};

HDOverlay.propTypes = {
    bgColor: PropTypes.string,
    opacity: PropTypes.number,
    animationDuration: PropTypes.number
};

HDOverlay.defaultProps = {
    bgColor: '#011831',
    opacity: 0.9,
    animationDuration: 1
};

export default HDOverlay;
