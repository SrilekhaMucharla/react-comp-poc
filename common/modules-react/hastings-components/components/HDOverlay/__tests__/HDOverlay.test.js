import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HDOverlay from '../HDOverlay';

Enzyme.configure({ adapter: new Adapter() });

const setup = (props) => {
    return shallow(<HDOverlay {...props} />);
};

describe('<HDOverlay />', () => {
    it('matches the snapshot', () => {
        const component = setup({ opacity: 0.9, color: 'black' });
        expect(component).toMatchSnapshot();
    });
});
