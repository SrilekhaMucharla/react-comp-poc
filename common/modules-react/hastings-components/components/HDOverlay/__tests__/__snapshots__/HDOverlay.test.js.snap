// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDOverlay /> matches the snapshot 1`] = `
<styled.div
  animationDuration={1}
  color="#011831"
  opacity={0.9}
/>
`;
